ricardopacheco/auction-fun-api
# [AFA-5] Create API with Restful structure

Issue number: resolves #5 

## Summary :red_circle:

The idea is to create a versioned rest-style API.

## Proposed / Possible solution :red_circle:

- Follow the rest API standards and conventions already existing in the ruby on rails literature.

## how to test :policeman:

Start application services with overmind with `overmind s`: After that, 

Start the related services, and open the console with the command `bin/console`. After that, in another tab create a cURL for the auction listing.

```ruby
curl http://localhost:5000/v1/auctions
```

> So far, we have not configured a fixed port to run Puma. Therefore, check if puma is actually running on port 5000.

## Risks / Impacts :red_circle:

- None anticipated

## Requirements for deployment :red_circle:

- Execute the deployment of changes in project X as a preliminary step before starting this deployment.
- Execute the migrate command before starting the deployment process.
